activedirectory.sgnl.yaml

```
@@ -0,0 +1,390 @@
+        externalId: dNSHostName
+        type: String
+      - name: servicePrincipalName
+        externalId: servicePrincipalName
+        type: String
+        list: true
+      - name: objectCategory
+        externalId: objectCategory
+        type: String
+      - name: isCriticalSystemObject
+        externalId: isCriticalSystemObject
+        type: Bool
+      - name: dSCorePropagationData
+        externalId: dSCorePropagationData
+        type: DateTime
+        list: true
+      - name: objectSid
+        externalId: objectSid
+        type: String
+      - name: msDSSupportedEncryptionTypes
+        externalId: msDS-SupportedEncryptionTypes
+        type: Int64
+      - name: lastLogonTimestamp
+        externalId: lastLogonTimestamp
+        type: Int64
+  Group:
+    description: "An entity representing a Group in an Active Directory"
+    displayName: Group
+    externalId: Group
+    pageSize: 100
+    pagesOrderedById: false
+    attributes:
+      - name: objectGUID
+        externalId: objectGUID
+        description: The objectGUID of the Group
+        type: String
+        indexed: true
+        uniqueId: true
+        attributeAlias: groupGUID
+      - name: dn
+        externalId: dn
+        description: The Distinguished Name (DN) of the group
+        type: String
+      - name: objectClass
+        externalId: objectClass
+        type: String
+        list: true
+      - name: cn
+        externalId: cn
+        type: String
+      - name: description
+        externalId: description
+        type: String
+      - name: distinguishedName
+        externalId: distinguishedName
+        type: String
+      - name: instanceType
+        externalId: instanceType
+        type: Int64
+      - name: whenCreated
+        externalId: whenCreated
+        type: DateTime
+      - name: whenChanged
+        externalId: whenChanged
+        type: DateTime
+      - name: uSNCreated
+        externalId: uSNCreated
+        type: Int64
+      - name: uSNChanged
+        externalId: uSNChanged
+        type: Int64
+      - name: name
+        externalId: name
+        type: String
+      - name: sAMAccountName
+        externalId: sAMAccountName
+        type: String
+      - name: sAMAccountType
+        externalId: sAMAccountType
+        type: Int64
+      - name: groupType
+        externalId: groupType
+        type: Int64
+      - name: objectCategory
+        externalId: objectCategory
+        type: String
+      - name: dSCorePropagationData
+        externalId: dSCorePropagationData
+        type: DateTime
+      - name: objectSid
+        externalId: objectSid
+        type: String
+  GroupMember:
+    description: "Group Member Entity in Active Directory"
+    displayName: GroupMember
+    externalId: GroupMember
+    pageSize: 100
+    pagesOrderedById: false
+    attributes:
+      - name: objectGUID
+        externalId: objectGUID
+        type: String
+        indexed: true
+        uniqueId: true
```

As discussed previously, `objectGUID` would be misleading here as this attribute is a composite-id generated from two other attributes. Please take care while naming attributes to prevent any confusion while writing policies.
```suggestion
      - name: id
        externalId: id
        type: String
        indexed: true
        uniqueId: true
```

---

To avoid hard-coded attribute, this will be targeted in improvements for adapter, as generic attributes are not currently supported